src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -890,32 +890,59 @@ def getVariablesFromMessage(String message) {
     return messageVars
 }
 /**
- * Watch for messages
+ * Watch for messages and verify their contents
  * @param msg_provider jms-messaging message provider
  * @param message trigger message
  */
 def watchForMessages(String msg_provider, String message) {
 
     def messageVars = getVariablesFromMessage(message)
 
-    topicsToWatchFor = ["org.centos.stage.ci.pipeline.package.running",
-                        "org.centos.stage.ci.pipeline.package.complete",
-                        "org.centos.stage.ci.pipeline.compose.running",
-                        "org.centos.stage.ci.pipeline.compose.complete",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.queued",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.running",
-                        "org.centos.stage.ci.pipeline.compose.test.integration.complete",
-                        "org.centos.stage.ci.pipeline.complete"]
-
-    topicsToWatchFor.each {
-        echo "Waiting for topic : " + it
+    messageContentValidationMap = [:]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.package.running'] =
+            ["branch", "repo"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.package.complete'] =
+            ["branch", "repo", "original_spec_nvr"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.compose.running'] =
+            ["branch", "repo"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.compose.complete'] =
+            ["branch", "repo", "compose_url"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.compose.test.integration.queued'] =
+            ["branch", "repo"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.compose.test.integration.running'] =
+            ["branch", "repo"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.compose.test.integration.complete'] =
+            ["branch", "repo", "compose_url"]
+    messageContentValidationMap['org.centos.stage.ci.pipeline.complete'] =
+            ["branch", "repo"]
+
+    messageContentValidationMap.each { k, v ->
+        echo "Waiting for topic : " + k
```

Comment:
Nit: couldn't we use interpolation to avoid a litany of "+" in strings?

Reply:
updated. tx